lib/gui/modules/drive-scanner.js

```diff
@@ -52,7 +53,9 @@ driveScanner.factory('DriveScannerService', (SettingsModel) => {
 
         return drive;
       });
+    })
 
+    .map((drives) => {
```

Comment:
`.flatMap` is a utility from RxJS to make all the `drivelist` function calls originate from the same stream (otherwise RxJS creates a new one for each).

Regarding the other `.map()`, I divided it into two to hopefully make the transformations clearer. The first one updates the descriptions, and the other one applies the system filtering. Happy to bring it back to one if that was better.